frontend/src/Components/GroupInfoManagement.tsx

```diff
@@ -59,6 +76,110 @@ export const GroupInfoManagement = ({
     onOpen();
   };
 
+  const useDownload = () => {
+    const downloader = useGetSpecificFile(
+      currentMachine,
+      currentFile,
+      (response) => {
+        validatedElim(response, {
+          success: async (r) => {
+            // here the response should have given an attached file through the attachment header stuff
+            // let's download
+
+            if (r.status == 200) {
+              const downloadLink = document.createElement("a");
+              downloadLink.href = URL.createObjectURL(
+                new Blob([await r.arrayBuffer()]),
+              );
+              downloadLink.download = currentFile;
+              downloadLink.click();
+            }
+          },
+          failure: () => {
+            // alert("Failure");
```

Comment:
Is there a chance we could create a `toast` here to inform the user that something went wrong? (see `useAddMachine` in `Hooks/Hooks.tsx` for an example)

EDIT: Oops, forgot the semantics of `useAuthFetch`. An invalid response is still considered a `success` so the toast should be above (in the case the `status != 200` but in any case - `toast`s on failed requests are probably a good idea.